rules/type-declaration/src/Rector/FunctionLike/ReturnTypeDeclarationRector.php

```diff
@@ -248,6 +256,27 @@ private function addReturnType(FunctionLike $functionLike, Node $inferredReturnN
         }
     }
 
+    private function isParentInVendor(ClassMethod $classMethod): bool
+    {
+        $parent = $classMethod->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parent instanceof Class_) {
+            return false;
+        }
+
+        if (! $parent->extends instanceof FullyQualified) {
+            return false;
+        }
+
+        $parentName = $parent->extends->toString();
+        $reflectionClass = new ReflectionClass($parentName);
+        if ($reflectionClass->isInternal()) {
+            return true;
+        }
+
+        $fileName = $reflectionClass->getFileName();
+        return Strings::contains((string) $fileName, 'vendor');
+    }
```

Comment:
This logic should be extracted to `packages/vendor-locker`

There might be some useful class like `Rector\VendorLocker\NodeVendorLocker\ClassMethodVendorLockResolver`

Reply:
implemented with use `VendorLockResolver->isReturnChangeVendorLockedIn()`